src/safeds/ml/nn/_model.py

```diff
@@ -0,0 +1,250 @@
+            accuracies.append(np.mean(tmp_accuracies))
+        copied_model._is_fitted = True
+        return copied_model
+
+    def predict(self, test_data: Table) -> TaggedTable:
+        """
+        Make a prediction for the given test data.
+
+        The original Model is not modified.
+
+        Parameters
+        ----------
+        test_data : Table
+            The data the network should predict.
+
+        Returns
+        -------
+        TaggedTable
+            The given test_data with an added "prediction" column at the end
+
+        Raises
+        ------
+        ModelNotFittedError
+            If the model has not been fitted yet
+        """
+        if not self._is_fitted:
+            raise ModelNotFittedError
+        copied_model = copy.deepcopy(self)
+        dataloader = test_data.into_dataloader(copied_model._batch_size)
+        copied_model._model.eval()
+        predictions = []
+        with torch.no_grad():
+            for x in dataloader:
+                predictions.append(copied_model._model(x).item())
+        return test_data.add_column(Column("prediction", predictions)).tag_columns("prediction")
+
+    @property
+    def is_fitted(self) -> bool:
+        """
+        Check if the model is fitted.
+
+        Returns
+        -------
+        is_fitted : bool
+            Whether the model is fitted.
+        """
+        return self._is_fitted
+
+
+class ClassificationNeuralNetwork:
+    def __init__(self, layers: list):
+        self._model = _PytorchModel(layers, is_for_classification=True)
+        self._batch_size = 1
+        self._is_fitted = False
+
+    def fit(self, train_data: TaggedTable, epoch_size: int = 25, batch_size: int = 1) -> Self:
+        """
+        Train the neural network with given training data.
+
+        The original model is not modified.
+
+        Parameters
+        ----------
+        train_data : TaggedTable
+            The data the network should be trained on.
+        epoch_size : int
+            The number of times the training cycle should be done
+        batch_size : int
+            The size of data batches that should be loaded at one time.
+
+        Raises
+        ------
+        ValueError
+            If epoch_size < 1
+            If batch_size < 1
+
+        Returns
+        -------
+        ClassificationNeuralNetwork
+            The trained Model
+        """
+        if epoch_size < 1:
+            raise OutOfBoundsError(actual=epoch_size, name="epoch_size", lower_bound=ClosedBound(1))
+        if batch_size < 1:
+            raise OutOfBoundsError(actual=batch_size, name="batch_size", lower_bound=ClosedBound(1))
+        copied_model = copy.deepcopy(self)
+        copied_model._batch_size = batch_size
+        dataloader = train_data.into_dataloader(copied_model._batch_size)
+
+        loss_fn = nn.BCELoss()
+
+        optimizer = torch.optim.SGD(copied_model._model.parameters(), lr=0.05)
+
+        loss_values = []
+        accuracies = []
```

Comment:
There's also no need to keep track of a single metric if the callback are available. -> Remove `accuracies` completley.